packages/venia-concept/src/components/Input/input.js

```diff
@@ -82,52 +111,83 @@ class Input extends Component {
         return required ? <div className={classes.requiredSymbol} /> : null;
     }
 
+    get resetButton() {
+        const { classes } = this.props;
+        const { name, attrs } = resetButtonIcon;
+        return (
+            <button
+                type="button"
+                className={classes.resetInput}
+                onClick={this.resetValue}
+            >
+                <Icon name={name} attrs={attrs} />
+            </button>
+        );
+    }
+
+    resetValue = () => {
+        const { fieldApi } = this.props;
+        fieldApi.setValue(null);
+    };
+
     render() {
-        const { helpText, labelText, requiredSymbol, rootClass } = this;
+        const {
+            helpTextSection,
+            labelText,
+            requiredSymbol,
+            rootClass,
+            resetButton
+        } = this;
         const {
             classes,
             placeholder,
             type,
             disabled,
-            required,
             title,
-            initialValue
+            initialValue,
+            field,
+            fieldApi,
+            fieldState
         } = this.props;
-        let { autoComplete, field } = this.props;
+        let { autoComplete } = this.props;
 
-        if (!this.state.dirty) {
-            field = initialValue ? initialValue : field;
-        }
         autoComplete = !autoComplete ? 'off' : autoComplete;
 
         return (
             <div className={rootClass}>
                 <span className={classes.label}>
                     {requiredSymbol} {labelText}
                 </span>
-                <Text
-                    initialValue={initialValue}
-                    className={classes.input}
-                    placeholder={placeholder}
-                    type={type}
-                    disabled={disabled}
-                    required={required}
-                    title={title}
-                    autoComplete={autoComplete}
-                    onChange={this.handleChange}
-                    onFocus={this.focusTextInput}
-                    onBlur={this.blurTextInput}
-                    field={field}
-                />
-                {helpText}
+                <div className={classes.inputContainer}>
+                    <BasicText
```

Comment:
I think we should have the option to pass in any `Informed` component, just wrapping `BasicText` really limits what we can do with `Informed`. This `Input` component could be made a lot more generic. 

For example, I think that the State field should use the `Select` component from `Informed` instead. Removes the need to have that error display too.